lib/Logger/LogAppenderSyslog.cpp

```diff
@@ -41,6 +41,32 @@ using namespace arangodb;
 
 using namespace arangodb::basics;
 
+namespace {
+#if defined(__clang__)
+#if __has_warning("-Wwritable-strings")
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wwritable-strings"
+#endif
+#endif
+auto findSyslogFacilityByName(std::string facility) -> int {
+  for (auto i = size_t{0}; i < LOG_NFACILITIES; ++i) {
+    if (strcmp(facilitynames[i].c_name, facility.c_str()) == 0) {
+      return facilitynames[i].c_val;
+    }
+    if (facilitynames[i].c_name == nullptr) {
+      return -1;
+    }
```

Comment:
I guess these two `if`s should be exchanged.
```suggestion
    if (facilitynames[i].c_name == nullptr) {
      return -1;
    }
    if (strcmp(facilitynames[i].c_name, facility.c_str()) == 0) {
      return facilitynames[i].c_val;
    }
```